twine/repository.py

```diff
@@ -153,6 +153,11 @@ def upload(self, package, max_redirects=5):
         return resp
 
     def package_is_uploaded(self, package, bypass_cache=False):
+        # NOTE(sigmavirus24): Not all indices are PyPI and pypi.io doesn't
+        # have a similar interface for finding the package versions.
+        if not self.url.startswith('https://pypi.python.org/pypi/'):
```

Comment:
So... I was under the impression that the call to `{index_url}/{package_safe_name}/json` only worked on Legacy PyPI. Does that mean we should change this as you suggest and then change L168 to:

```py
url = '{url}/{safe_name}/json'.format(url=self.url, safe_name=safe_name)
```

Or is it just that uploading to pypi.io can/should still talk to Legacy PyPI?